### DISCUSSION TITLE
Quantization example code

### DISCUSSION THREAD

**Discussion by simonmurrell1979** [SixLabors/ImageSharp#2265] [Q&A] [answered]
Hello

I hope you are doing well. Is there any example code on loading a PNG image and applying ImageSharp quantization to a PNG file?

I have been using PngQuant to compress images but am looking for a better solution and came across this library. Saw example images below.

https://jimbobsquarepants.github.io/wudemo/

**Comment by JimBobSquarePants** [comment, marked as answer]
Sure...

```c#
string inPath = "...";
string outPath = "...";
using Image image = Image.Load(inPath);
image.Save(outPath, new PngEncoder() { ColorType = PngColorType.Palette, BitDepth = PngBitDepth.Bit8 });
```

There are overloads for Load/Save that take streams also.

**Reply by simonmurrell1979** [reply]
I am trying to make the image smaller like your WuDemo. Not sure if I am doing it correctly or not?

using (SixLabors.ImageSharp.Image image = SixLabors.ImageSharp.Image.Load(@"C:\Users\SimonM\Desktop\Temp\input.png"))
            {

                var quantizer = new WuQuantizer(new QuantizerOptions 
                { 
                    Dither = null, 
                    MaxColors = 256 
                });

                image.Mutate(x => x.Quantize(quantizer));

                image.Save(@"C:\Users\SimonM\Desktop\Temp\output.png",
                    new PngEncoder() { ColorType = PngColorType.Palette, BitDepth = PngBitDepth.Bit8 });
                image.Save(@"C:\Users\SimonM\Desktop\Temp\output2.png");

            }

**Reply by JimBobSquarePants** [reply]
My sample encodes the image as an 8 bit palette png that is quantized using the Wu Quantizer by default. 

There’s no need to explicitly quantize and you can pass your own quantizer to the encoder

**Reply by simonmurrell1979** [reply]
Ah okay. I see. Thank you for the information. I want confused as the file was ending up being larger than the input. It also helps if you given the correct input file to test with ;)
